main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -31,9 +31,10 @@ object Desugar {
       case (k, v) => visitUnit(k, v)
     }
 
-    val allUnits = units.foldLeft(Map.empty[Ast.Source, DesugaredAst.CompilationUnit]) {
-      case (macc, (k, v)) => macc + (k -> v)
-    }
+    val allUnits = ParOps.parAgg(units, Map.empty[Ast.Source, DesugaredAst.CompilationUnit])(
```

Comment:
Better to use parMap!

Reply:
I think this is completely redundant. It's just a fold at the end, no? See latest version

Reply:
Should be mapValues I think